Upstream-Status: Pending

Fix the following compilation issues with gcc 4.6.0

| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_updateinfoxml.c: In function 'startElement':
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_updateinfoxml.c:278:25: error: variable 'status' set but not used [-Werror=unused-but-set-variable]
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_updateinfoxml.c:386:36: error: variable 'src' set but not used [-Werror=unused-but-set-variable]
| cc1: all warnings being treated as errors
| 
| make[2]: *** [ext/CMakeFiles/satsolverext.dir/repo_updateinfoxml.o] Error 1
| make[2]: *** Waiting for unfinished jobs....
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/src/transaction.c: In function 'transaction_add_obsoleted':
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/src/transaction.c:1933:21: error: variable 'oldcount' set but not used [-Werror=unused-but-set-variable]
| cc1: all warnings being treated as errors
| 
| make[2]: *** [src/CMakeFiles/satsolver.dir/transaction.o] Error 1
| make[2]: Leaving directory `/build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git'
| make[1]: *** [src/CMakeFiles/satsolver.dir/all] Error 2
| make[1]: *** Waiting for unfinished jobs....
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_rpmdb.c: In function 'rpm_byfp':
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_rpmdb.c:2406:20: error: variable 'headerend' set but not used [-Werror=unused-but-set-variable]
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_rpmdb.c: In function 'parsekeydata':
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_rpmdb.c:2833:29: error: variable 'scr' set but not used [-Werror=unused-but-set-variable]
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_rpmdb.c:2669:7: error: variable 'useridl' set but not used [-Werror=unused-but-set-variable]
| /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git/ext/repo_rpmdb.c:2667:7: error: variable 'pubkeyl' set but not used [-Werror=unused-but-set-variable]
| cc1: all warnings being treated as errors
| 
| make[2]: *** [ext/CMakeFiles/satsolverext.dir/repo_rpmdb.o] Error 1
| make[2]: Leaving directory `/build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git'
| make[1]: *** [ext/CMakeFiles/satsolverext.dir/all] Error 2
| make[1]: Leaving directory `/build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/git'
| make: *** [all] Error 2
| FATAL: oe_runmake failed
| ERROR: Function 'do_compile' failed (see /build_disk/poky_build/build_gcc_4.6.0/tmp/work/qemux86-poky-linux/sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9/temp/log.do_compile.21830 for further information)
NOTE: package sat-solver-0.0-git1+0a7378d5f482f477a01cf1690d76871ab8bdcc32-r9: task do_compile: Failed

Nitin A Kamble <nitin.a.kamble@intel.com> 2011/05/05

Index: git/src/transaction.c
===================================================================
--- git.orig/src/transaction.c
+++ git/src/transaction.c
@@ -1930,7 +1930,7 @@ transaction_add_obsoleted(Transaction *t
   Repo *installed = pool->installed;
   Id p;
   Solvable *s;
-  int i, j, k, max, oldcount;
+  int i, j, k, max;
   Map done;
   Queue obsq, *steps;
 
@@ -1945,7 +1945,6 @@ transaction_add_obsoleted(Transaction *t
     return;
   /* make room */
   steps = &trans->steps;
-  oldcount = steps->count;
   queue_insertn(steps, 0, max);
 
   /* now add em */
Index: git/ext/repo_updateinfoxml.c
===================================================================
--- git.orig/ext/repo_updateinfoxml.c
+++ git/ext/repo_updateinfoxml.c
@@ -275,7 +275,7 @@ startElement(void *userData, const char 
        */
     case STATE_UPDATE:
       {
-	const char *from = 0, *status = 0, *type = 0, *version = 0;
+	const char *from = 0, __attribute__((__unused__)) *status = 0, *type = 0, *version = 0;
 	for (; *atts; atts += 2)
 	  {
 	    if (!strcmp(*atts, "from"))
@@ -383,7 +383,7 @@ startElement(void *userData, const char 
        */
     case STATE_PACKAGE:
       {
-	const char *arch = 0, *name = 0, *src = 0;
+	const char *arch = 0, *name = 0, __attribute__((__unused__)) *src = 0;
 	Id evr = makeevr_atts(pool, pd, atts); /* parse "epoch", "version", "release" */
 	Id n, a = 0;
 	Id rel_id;
Index: git/ext/repo_rpmdb.c
===================================================================
--- git.orig/ext/repo_rpmdb.c
+++ git/ext/repo_rpmdb.c
@@ -2403,7 +2403,7 @@ void *
 rpm_byfp(FILE *fp, const char *name, void **statep)
 {
   struct rpm_by_state *state = *statep;
-  int headerstart, headerend;
+  int headerstart, __attribute__((__unused__))headerend;
   RpmHead *rpmhead;
   int sigdsize, sigcnt, l;
   unsigned char lead[4096];
@@ -2664,9 +2664,9 @@ parsekeydata(Solvable *s, Repodata *data
   unsigned char keyid[8];
   unsigned int kcr = 0, maxex = 0;
   unsigned char *pubkey = 0;
-  int pubkeyl = 0;
+  int __attribute__((__unused__)) pubkeyl = 0;
   unsigned char *userid = 0;
-  int useridl = 0;
+  int __attribute__((__unused__)) useridl = 0;
 
   for (; pl; p += l, pl -= l)
     {
@@ -2830,7 +2830,7 @@ parsekeydata(Solvable *s, Repodata *data
 	    {
 	      int j, ql, haveissuer;
 	      unsigned char *q;
-	      unsigned int ex = 0, scr = 0;
+	      unsigned int ex = 0, __attribute__((__unused__))scr = 0;
 	      unsigned char issuer[8];
 
 	      // printf("V4 signature packet\n");
